kube-client/src/client/discovery_types.rs

```diff
@@ -0,0 +1,340 @@
+        let json = r#"{
+            "kind": "APIGroupDiscoveryList",
+            "apiVersion": "apidiscovery.k8s.io/v2",
+            "metadata": {},
+            "items": [
+                {
+                    "metadata": {
+                        "name": "apps"
+                    },
+                    "versions": [
+                        {
+                            "version": "v1",
+                            "freshness": "Current",
+                            "resources": [
+                                {
+                                    "resource": "deployments",
+                                    "responseKind": {
+                                        "group": "apps",
+                                        "version": "v1",
+                                        "kind": "Deployment"
+                                    },
+                                    "scope": "Namespaced",
+                                    "singularResource": "deployment",
+                                    "verbs": ["create", "delete", "deletecollection", "get", "list", "patch", "update", "watch"],
+                                    "shortNames": ["deploy"],
+                                    "categories": ["all"],
+                                    "subresources": [
+                                        {
+                                            "subresource": "status",
+                                            "responseKind": {
+                                                "group": "apps",
+                                                "version": "v1",
+                                                "kind": "Deployment"
+                                            },
+                                            "verbs": ["get", "patch", "update"]
+                                        },
+                                        {
+                                            "subresource": "scale",
+                                            "responseKind": {
+                                                "group": "autoscaling",
+                                                "version": "v1",
+                                                "kind": "Scale"
+                                            },
+                                            "verbs": ["get", "patch", "update"]
+                                        }
+                                    ]
+                                }
+                            ]
+                        }
+                    ]
+                }
+            ]
+        }"#;
+
+        let result: APIGroupDiscoveryList = serde_json::from_str(json).unwrap();
+
+        assert_eq!(result.items.len(), 1);
+
+        let apps_group = &result.items[0];
+        assert_eq!(
+            apps_group.metadata.as_ref().and_then(|m| m.name.as_ref()),
+            Some(&"apps".to_string())
+        );
```

Comment:
Thinking out loud: This is more awkward than regular `Resource` implementors, but looking upstream we don't have an `impl Resource for APIVersionDiscovery` [in k8s-pb](https://docs.rs/k8s-pb/latest/k8s_pb/api/apidiscovery/v2/struct.APIVersionDiscovery.html) either, and i guess it would be misleading to mark it as a `Resource` when we cannot use it in an `Api`. So this is probably fine. Maybe a follow-up issue on some convenience helpers makes sense, but this is just ideas for the future.